tests/test_purl_spec.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) the purl authors
+# SPDX-License-Identifier: MIT
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Visit https://github.com/package-url/packageurl-python for support and
+# download.
+
+import json
+import os
+
+import pytest
+
+from packageurl import PackageURL
+
+current_dir = os.path.dirname(__file__)
+root_dir = os.path.abspath(os.path.join(current_dir, ".."))
+spec_file_path = os.path.join(root_dir, "spec", "tests", "spec", "specification-test.json")
+
+valid_purl_types_file = os.path.join(root_dir, "spec", "purl-types-index.json")
+
+
+with open(spec_file_path, "r", encoding="utf-8") as f:
+    test_cases = json.load(f)
+
+with open(valid_purl_types_file, "r", encoding="utf-8") as f:
+    valid_purl_types = json.load(f)
```

Comment:
Are we using `valid_purl_types` any where in test?

Reply:
Good catch, previously thinking to use it, but not needed. Thanks!